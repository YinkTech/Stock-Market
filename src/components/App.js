import React from 'react';
import { TrendList } from '../containers/TrendList';
import { MarketList } from './../containers/MarketList';
import './../styles/main.css'


const App = () => {
  return (
    <div className="App d-md-flex">
      <MarketList className=" p-2" />
      <TrendList />
    </div>
  );
}

export default App;

       