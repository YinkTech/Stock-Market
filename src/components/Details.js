import React, { useEffect } from 'react'
import { useParams } from 'react-router-dom'
import { Link } from 'react-router-dom';
import axios from 'axios';
import { useDispatch, useSelector } from 'react-redux';
import { selectedMarket, removeSelectedMarket } from '../redux/actions/action';
import './../styles/details.css'

export const Details = () => {
  const market = useSelector((state) => state.selectedMarketReducer);
  const {image, name, symbol, sentiment_votes_up_percentage, sentiment_votes_down_percentage, market_cap_rank, last_updated, community_score, liquidity_score, public_interest_score, genesis_date, description, community_data, links} = market;
  const { marketId } = useParams();
  const dispatch = useDispatch();

  const fetchMarketDetail = async() => {
    const response = await axios
    .get(`https://api.coingecko.com/api/v3/coins/${marketId}?localization=false&tickers=false&market_data=false&community_data=true&developer_data=false&sparkline=false`)
    .catch((err) => {
      console.log("Err ", err);
    });
    dispatch (selectedMarket(response.data));
  };

  useEffect(() => {
    if (marketId && marketId !== "" ) fetchMarketDetail();
    return () => {
      dispatch(removeSelectedMarket());
    };
  }, [marketId])

  return (
    <div className='details'>
      <div className=''>
        {Object.keys(market).length === 0 ? (
          <div className='text-info mx-5 fs-2 text-uppercase'>loading</div>
              ) : (
          <div className='d-md-flex justify-content-between'>
            <div className='det-first-sec'>
              <Link to='/'>
                <button className="btn text-info d-block fw-bolder fs-3 px-2 py-0 text-capitalize"> <i className="bi bi-skip-backward"></i> </button>
              </Link>
              <hr className='separator' />
              <div className='p-2'>
                <div className='mb-5 px-3 d-flex align-items-center'>
                <img src={image.large} alt="img" className='img-fluid details-img m-2' />
                  <div className='d-flex fs-1 flex-column'>
                    <span className='text-warning'>{name}</span>
                    <span className='text-muted'>{symbol}</span>
                  </div>
                </div>

                <div className='px-2 mt-5'>
                  <div className=''>
                    <h3 className='fw-bolder fs-3'>
                      Key <span className='def-color'>Statistics</span> 
                    </h3>
                    <div className='text-center p-2'>
                      <div className='mb-4 mx-3 market-row p-3 shadow text-success'>
                        <span className='info'>sent votes up </span>
                        <span>{sentiment_votes_up_percentage}</span>
                      </div>
                      <div className=' mb-4 mx-3 market-row p-3 shadow text-danger'>
                        <span className='info d-inline-block'>sent votes down </span>
                        <span>{sentiment_votes_down_percentage}</span>
                      </div>
                      <div className=' mb-4 mx-3 market-row p-3 shadow'>
                        <span className='info'>Market Rank </span>
                        <span>{market_cap_rank}</span>
                      </div>
                      <div className=' mb-4 mx-3 market-row p-3 shadow text-info'>
                        <span className='info'>Community Score</span>
                        <span>{community_score}</span>
                      </div>

                        <hr className='separator' />

                      <div className=' mb-4 mx-3 market-row p-3 shadow text-primary'>
                        <span className='info'>liquidity Score </span>
                        <span>{liquidity_score}</span>
                      </div>
                      <div className=' mb-4 mx-3 market-row p-3 shadow text-warning'>
                        <span className='info'> Public Interest Sscore </span>
                        <span>{public_interest_score}</span>
                      </div>
                      <div className=' mb-4 mx-3 market-row p-3 shadow'>
                        <span className='info'>Last Update </span>
                        <span>{last_updated}</span>
                      </div>
                      <div className='mb-4 mx-3 market-row p-3 shadow text-info'>
                        <span className='info'>genesis_date</span>
                        <span>{genesis_date}</span>
                      </div>
                    </div>
                  </div>
                  <div className=''>
                    <h3 className='fw-bolder text-muted fs-3'>
                      Social
                    </h3>

                    <div className=' text-muted my-2'>
                      <div className='text-capitalize w-100 justify-content-between d-inline-flex'>
                        <span><i className="bi bi-code-slash"></i><i className="bi bi-code-slash"></i><i className="bi bi-code-slash"></i></span>
                        <span className='text-end'>subscriber</span>
                        <span>Link</span>
                      </div>
                      <div className='text-capitalize w-100 justify-content-between d-inline-flex'>
                        <span>twitter</span>
                        <span className='text-end'>{community_data.twitter_followers}</span>
                        <span><a href='https://twitter.com/{links.twitter_screen_name}'><i className="bi bi-twitter"></i></a></span>
                      </div>
                      <div className='text-capitalize w-100 justify-content-between d-inline-flex'>
                        <span>reddit</span>
                        <span className='text-end'>{community_data.reddit_subscribers}</span>
                        <span><a href='https://reddit.com/{links.reddit_screen_name}'><i className="bi bi-reddit"></i></a></span>
                      </div>
                      <div className='text-capitalize w-100 justify-content-between d-inline-flex'>
                        <span>Facebook</span>
                        <span className='text-end'>{community_data.facebook_likes}</span>
                        <span><a href='https://facebook.com/{links.facebook_screen_name}'><i className="bi bi-facebook"></i></a></span>
                      </div>
                      <div className='text-capitalize w-100 justify-content-between d-inline-flex'>
                        <span>homepage</span>
                        <span><a href={links.homepage[0]}><i className="bi bi-link"></i></a></span>
                        
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div className='aside-detail p-2'>
              <h4 className='fs-3 fw-bolder'>Coin Profile</h4>
              <h6 className='text-info sr-only sr-only-focusable'>{name}</h6>
              <p className='details-text' > {description.en} </p>
            </div>
          </div>
        )}
      </div>
    </div>
  );
};
