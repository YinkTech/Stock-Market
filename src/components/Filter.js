import React from 'react'
import { PropTypes } from 'prop-types';
import { Dropdown } from 'react-bootstrap';

const Filter = ({handleFilter}) => (
  <Dropdown className='mb-3' onSelect={handleFilter}>
  <Dropdown.Toggle className="dropdown bg-success">
    Page Filter
  </Dropdown.Toggle>

  <Dropdown.Menu className=" bg-info text-center w-auto">
    <Dropdown.Item eventKey="1" className='border fw-bold fs-3'>1</Dropdown.Item>
    <Dropdown.Item eventKey="2" className='border fw-bold fs-3'>2</Dropdown.Item>
    <Dropdown.Item eventKey="3" className='border fw-bold fs-3'>3</Dropdown.Item>
    <Dropdown.Item eventKey="7" className='border fw-bold fs-3'>4</Dropdown.Item>
    <Dropdown.Item eventKey="8" className='border fw-bold fs-3'>5</Dropdown.Item>
  </Dropdown.Menu>
</Dropdown>
);


Filter.propTypes = {
  handleFilter: PropTypes.func.isRequired,
};

export default Filter;