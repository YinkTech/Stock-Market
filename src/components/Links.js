import React from 'react';
import { Link } from 'react-router-dom';
import '../styles/navBar.css'

export  const Links = () => {
  return (
      <div className='links d-none d-sm-flex p-2' id='my-aside'>
        <Link to="/Stock-Market" className='home'>
           <i className="bi bi-house me-1"></i>
            Home
        </Link>
      </div>
  );
};
