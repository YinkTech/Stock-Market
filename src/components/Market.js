import React from 'react'
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';

export const Market = () => {
  
  const markets = useSelector((state) => state.marketReducer.markets);
  const renderList = markets.map((market) => {
  const formatToCurrency = amount => {
      return "$ " + amount.toFixed(2).replace(/\d(?=(\d{3})+\.)/g, "$&,");
  };
  const { id, image, name, symbol , current_price, low_24h } = market;
    return (
      <div key={id} className='mb-4 mx-3 market-row p-3 shadow'>
          <Link className='text-white text-decoration-none' to={`/Details/${id}`} >
            <div>
              <div className='market-cont'>
                  <div className='d-flex mb-1'>
                    <img src={image} alt="logo" className='coin-img img-fluid  mx-2 me-5' />
                    <div className=' d-flex flex-column'><b>{name}</b> <b className='text-info'>[{symbol}]</b></div>
                  </div>
                  <div className='px-4 mt-2 py-2 text-center'>
                     <span className='d-flex align-items-center justify-content-between'>Today&apos;s Price <small className='text-color'><i className="bi bi-arrow-up-right"></i> {formatToCurrency(current_price)} </small></span>
                      <hr className='separator' />
                      <span className='d-flex align-items-center justify-content-between'>Low price <small className='text-danger'><i className="bi bi-arrow-down-right"></i> {formatToCurrency(low_24h)}</small></span>
                    </div>
              </div>
            </div>
          </Link>
       </div>
    );
  });

  return <>{renderList}</>
};



