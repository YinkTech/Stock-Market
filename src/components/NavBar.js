import React from 'react';
import { Link } from 'react-router-dom';
import '../styles/navBar.css'

export const NavBar = () => {
  return (
    <>
      <div className='nav justify-content-center sticky-top bg-dark text-decoration-none text-white'>
            <div className='d-flex align-items-center' > <i className="bi bi-cash-coin fs-2 px-3"></i> <Link className='logo text-uppercase fs-2' to="/"> Stock Market </Link></div>
      </div>
   </>
  );
};
