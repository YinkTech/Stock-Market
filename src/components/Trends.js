import React from 'react'
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';

export const Trends = () => {
  const coins = useSelector((state) => state.trendReducer.coin);
  const renderTlist = coins.map((coin) => {
    const formatToCurr = amount => {
        return "$ " + amount.toFixed(2).replace(/\d(?=(\d{3})+\.)/g, "$&,");
    };
    const { id, large, name, symbol , price_btc, score } = coin.item;
      
    return (
        <div key={id} className='mb-4 mx-3 market-row p-3 shadow'>
            <Link className='text-white text-decoration-none' to={`/Details/${id}`} >
              <div>
                <div className='market-cont'>
                    <div className='d-flex mb-1'>
                      <img src={large} alt="logo" className='coin-img img-fluid  mx-2 me-5' />
                      <div className=' d-flex flex-column'><b>{name}</b> <b className='text-info'>[{symbol}]</b></div>
                    </div>
                    <div className='px-4 mt-2 py-2 text-center'>
                      <span className='d-flex align-items-center justify-content-between'>Today's Price <small className='text-color'><i className="bi bi-arrow-up-right"></i> {formatToCurr(price_btc)}</small></span>
                        <hr className='separator' />
                        <span className='d-flex align-items-center justify-content-between'>Score <small className='text-warning'> {score}</small></span>
                      </div>
                </div>
              </div>
            </Link>
        </div>
    );
  });

  return <>{renderTlist}</>
};



