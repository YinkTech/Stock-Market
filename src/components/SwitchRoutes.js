import React from 'react'
import { NavBar } from './NavBar'
import { App } from './App'
import { Details } from './Details'
import { Route, Routes } from 'react-router-dom'

export const SwitchRoutes = () => {
  return (
    <>
      <NavBar />
        <Routes>
          <Route path='/Stock-Market' element={<App />} exact />
          <Route path='/Details' element={<Details />} />
        </Routes>
    </>
  )
}

