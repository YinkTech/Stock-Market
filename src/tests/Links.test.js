
import renderer from 'react-test-renderer';
import { Links } from './../components/Links'
import { Link } from 'react-router-dom'

describe('Links component', () => {
  test('snapshot test', () => {
    const linksCheck = renderer.create(
      <links />
    )
    .toJSON();
      expect(linksCheck).toMatchSnapshot();
  });
});