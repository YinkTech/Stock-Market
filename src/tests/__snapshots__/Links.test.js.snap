// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links component snapshot test 1`] = `<links />`;
